language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -1799,11 +1827,23 @@ impl<'mm, 'up> FunctionContext<'mm, 'up> {
                     .llvm_cx
                     .named_struct_type(&struct_name)
                     .expect("no struct type");
+                debug!(target: "structs", "stype {}", stype.as_any_type().print_to_str().to_string());
                 let fvals = src
                     .iter()
                     .map(|i| (self.locals[*i].llty, self.locals[*i].llval))
                     .collect::<Vec<_>>();
+                for (idx, (ty, _)) in fvals.iter().enumerate() {
+                    let local_src_name = ty.print_to_str().to_string();
+                    debug!(target: "structs", "local_src_ty_{} {}", idx, local_src_name);
+                    for l in self.mty_to_llty.values() {
+                        if l.0 == ty.0 {
+                            debug!(target: "structs", "ty is struct ");
+                        }
+                    }
+                }
                 let dst_idx = dst[0];
+                let local_dst_name = self.locals[dst_idx].llty.print_to_str().to_string();
+                debug!(target: "structs", "local_dst_ty {}", local_dst_name);
                 let ldst = (self.locals[dst_idx].llty, self.locals[dst_idx].llval);
                 builder.insert_fields_and_store(&fvals, ldst, stype);
             }
```

Comment:
I changed the map to `llty_to_mty` which makes more sense.
My point was to demo that using this map makes these two pieces equivalent:
```diff
+                for (idx, (ty, _)) in fvals.iter().enumerate() {
+                   if self.llty_to_mty.get(ty).unwrap().is_struct() {
+                        let local_src_name = ty.print_to_str().to_string();
+                        debug!(target: "structs", "0: field {} of src {} is struct", idx, local_src_name);
+                    }
+                }
-                for (idx, src_mty, src_llty) in src
-                    .iter()
-                    .map(|i| (i, &self.locals[*i].mty, &self.locals[*i].llty))
-                    .collect::<Vec<(_,_,_)>>() {
-                    if src_mty.is_struct() {
-                        let local_src_name = src_llty.print_to_str().to_string();
-                        debug!(target: "structs", "1: field {} src type {} is struct", idx, local_src_name);
-                    }
-                }
```

Reply:
My idea was to add a map `llty -> mty` (or `mty -> llty`) since I anticipate we may not need to generate or allocate `llty` for each `mty`.
Using this map in `Operation::Pack` was to illustrate how it works.
Since I'm back to using `mty` from `locals`, there is no remaining usage of this map in this patch and the very patch becomes useless.
I will not merge it, but will keep in my fork for possible future references.

Reply:
I don't see the removal of the map you mentioned above. Did you commit to the branch used in for this PR?

Since the goal of this patch is printing debugging information, any future optimizations or utilities should come in with those patches. That said, the utility of the map even then is questionable (though that can be re-examined when the time comes). If we determine a local is unnecessary, we will never create the LLVM type in the first place-- indeed, we would mark the entry for the temp as dead in `FunctionContext::locals`-- as it would no longer be referenced.

Reply:
Map `llty_to_mty` is still there but it is not used anymore.

Please re-read the description, it mentions SSA as a **future** analysis and refers to printout of the provided debugging to demo that we may need it. 

The title of the PR says that it is NFC and for debugging only.

Reply:
Yes, I read the description carefully. This patch is (or should be) about debug info printing today. The map is not necessary for that task-- all the information needed is already there. Nor is that map necessary (I believe) for a future SSA analysis. 

Now if a future time comes where we indeed demonstrate the need that sort of map, it should be added and reviewed as part of _that_ future work. We do not want to introduce any unnecessary structures or code-- especially when there is no demonstrated immediate need.

As an aside, SSA analysis would be on stackless bytecode, not on LLVM IR. For the cases I believe you to be considering-- redundant allocas and loads/stores from/to them, one would never create such an LLVM type (or reference) in the first place-- it would not appear in any map or structure. Specifically, when populating FunctionContext::locals, we'd already have discovered (from analysis on hypothetical SSA-SBC) that local slot is not going to be used. It would be marked dead/poison and not touched again-- it would never have an LLVM type at all.

Reply:
Reiterating, this additional map and code using it is unnecessary. Nevertheless I'm approving the patch since we're at an impasse. In general, we should avoid adding complexity that does not have demonstrated need. Future need (should it arise) would ordinarily be added/reviewed as part of that future (orthogonal) work.

Reply:
This map allowed to link `mty` properties to `llty`. At some places it becomes very convenient, since testing `llty` properties we do thru `llvm-sys` and interfaces here are not very clear.
It is true that the chosen usage of this map is unnecessary, it can be done directly from existing at these point locals.
But the purpose of this was to demo that the information built in the map (it uses boxing and cloning) is correct.
As it is now, after I removed any use of this map, the PR got very insignificant value and I'm not merging it.